.github/workflows/check-labels.yml

```diff
@@ -0,0 +1,15 @@
+name: check-labels
+on:
+  pull_request:
+    types: [opened, reopened, labeled, unlabeled]
+  pull_request_review:
+    types: [submitted, dismissed]
+jobs:
+  check-labels:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: RIOT-OS/check-labels-action@v1.0.0
+      with:
+        access_token: ${{ secrets.GITHUB_TOKEN }}
+        unset_labels: 'CI: needs squashing,State: waiting for other PR'
```

Comment:
These 2 ones are already checked by Murdock, so this is a bit redundant.

Reply:
It is true that they are checked by Murdock, but I would not say it is redundant. For Murdock to block when one of these labels are set, the `CI: ready for build` label needs to be set and a complete Murdock run must pass. This GitHub action is reactive, so if for it marks the PR as failed as soon as one of those labels is set. Take https://github.com/RIOT-OS/RIOT/pull/14770 for example. Here https://github.com/RIOT-OS/RIOT/commit/4c21e2dc9539564d98f7a199ea5a33de46900e01 was already passed as succeeding in Murdock, when I ACK'd, giving the potential for an accidental merge. The only way to prevent that would have been to dismiss my review, requiring myself to come back again later, or set the `CI: needs squashing label` and re-trigger Murdock, potentially giving Murdock two runs on that PR (once for my re-triggering, once when @benpicco force-pushed). I knew @benpicco would squash fast enough that I did not have to do either, but such dilemma could be prevented in the future.

Reply:
(also when they are checked here, we could remove those checks from Murdock, were they are IMHO pretty unreliable, since they are checked when Murdock is triggered, not when the actual test is run, putting some time difference of potentially hours between the the two events)

Reply:
We should remove the label check from Murdock and go for the github actions.

Reply:
Agreed, but let's do that in a follow-up ;-).

Reply:
>  let's do that in a follow-up ;-)

Why, this is as simple as removing some lines in the pr_check.sh script ? And it could be done in one go.

Reply:
Because it touches completely different files that only incidentally are related to this PR... And again, I don't see the redundancy you see. Plus, there is no reason to not have both for a short of time, even if it were redundant.

Reply:
> Because it touches completely different files that only incidentally are related to this PR

Well, the `pr_check.sh` script contains code that check if labels are set. Which is exactly the same purpose of this PR. So I see this PR (for the part that checks if some labels are set) as replacement of this check. That's why I would remove it in the same PR.

Reply:
We rarely remove a replaced feature in the same PR as the replacing feature is introduced. Why should it be the case here? While it might be a clear cut decision, imagine there would be some discussion required on either of the two changes. If we put everything in one PR we effectively have to discuss two topics that could easily be done in two separate discussions. I think we should keep those discussions separate in either case.

Reply:
Besides, we start bike-shedding here...

Reply:
Another argument: if I remove the check here there is a (hopefully just a short but still significant) time window in which the labels are unchecked: Between merging this PR and somebody updating the branch protection for this new check to be required.

Reply:
> Between merging this PR and somebody updating the branch protection for this new check to be required.

Since you are the author of this PR, have enough rights to update the branch protection and are aware of this issue, you are in a good position to prevent that :)

Reply:
Sure, but I rather would make sure another lock is in place before removing the other. This is just concurrent programming 101...

Reply:
If this is a matter of trusting that the follow-up is provided: here you go https://github.com/RIOT-OS/RIOT/pull/14872.

Reply:
Ok, let's move forward with this one. Since #14872 is here there's no reason to block this longer.